marketplace/deployer_helm_tiller_base/bin/deploy_internal.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eox pipefail
+
+app_uid=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.metadata.uid}')
+app_api_version=$(kubectl get "applications/$NAME" \
+  --namespace="$NAMESPACE" \
+  --output=jsonpath='{.apiVersion}')
+
+for chart in /data/chart/*; do
+  # Expand and apply the template for the Application resource.
+  # Note: This must be done out-of-band because the Application resource
+  # is created before the deployer is invoked, but helm does not handle
+  # pre-existing resources.
+  helm template \
+      --name="$NAME" \
+      --namespace="$NAMESPACE" \
+      --values=<(print_config.py --output=yaml) \
+      --set "$(print_config.py --xtype APPLICATION_UID --key true)=" \
```

Comment:
Nit: Oversight from my part: `--key` argument should have an action of recording true when specified, so that we don't need to specify `true` here. Not blocking.

Reply:
+1 - I was so sad when I had to add `true`.

`print_config.py` is pending an overhaul anyways - will resolve at that time.